docs-client/src/containers/App/index.tsx

```diff
@@ -88,25 +91,35 @@ const styles = (theme: Theme) =>
 interface State {
   mobileDrawerOpen: boolean;
   specification?: Specification;
+  servicesState: boolean;
+  enumsState: boolean;
+  structsState: boolean;
+  exceptionsState: boolean;
 }
 
 type Props = WithStyles<typeof styles> & RouteComponentProps<{}>;
 
 interface AppDrawerProps extends WithStyles<typeof styles> {
   specification: Specification;
   navigateTo: (url: string) => void;
+  state: State;
```

Comment:
Pass in the four props individually to prevent unrelated state changes from affecting this component

Reply:
I tried to pass four props individually but turned out that the DOM will not refresh by value change (E.g. `servivesOpen: true` to `servivesOpen: false`), then click some redirection menu (one item in `Services`), the menu will be collapsed. 
But by passing them inside the `state`, this works.

Reply:
I checked out the branch and am trying to reproduce this, but can't find any problem. Can you describe in more detail with 1), 2), 3) etc what steps cause the four boolean approach to not work? Code-wise, I don't see any reason it wouldn't.

This is the code I'm trying and not finding problem with
https://github.com/anuraaga/armeria/blob/namidairo777-Apply_Collapse_to_side_menu/docs-client/src/containers/App/index.tsx#L116